documentation/01-Prerequisites.md

```diff
@@ -111,6 +81,113 @@ sudo apt-get install exuberant-ctags
 
 > When available `make ctags` can be invoked to help generate a `tags` database at the root of the source tree to allow easier code navigation in editors that support it.
 
+#### Docker
+
+To automatically setup a Linux build environment using `docker`, run the following:
+
+```sh
+# Select your target Ubuntu version:
+UbuntuVersion=20.04
+# Build the docker image:
+docker build --build-arg=UbuntuVersion=$UbuntuVersion -t mlos/build:ubuntu-$UbuntuVersion .
+```
+
+> Where `UbuntuVersion` can also be set to another supported version of Ubuntu.
+
+> Tip: you can also pass `--build-arg=http_proxy=http:/some-proxy-caching-host:3128` to help direct `apt` and `pip` to fetch the necessary packages via local caches.
+
+See [02-Build.md](./02-Build.md#docker) for instructions on how to run this image.
+
+
+### Clone the repository
+
+Cross platform
+
+```shell
+git clone https://github.com/microsoft/MLOS.git
+```
+
+> See <https://git-scm.com/book/en/v2/Getting-Started-Installing> for help installing `git`.
+
+
+### Linux Python Install
+
+1. Install Python 3.7
+
+    ```sh
+    # We need to add a special apt repository for Python 3.7 support:
+    sudo apt-get -y install software-properties-common apt-transport-https
+	sudo add-apt-repository -y ppa:deadsnakes/ppa
+	sudo apt-get update
+	sudo apt-get -y install python3.7
+    ```
+
+2. Install MLOS Python dependencies:
+
+    ```sh
+    # Also add some dependencies needed by some of the pip modules
+    sudo apt-get -y install python3-pip python3.7-dev \
+        build-essential libfreetype-dev unixodbc-dev
+    ```
+
+    ```sh
+    python3.7 -m pip install --upgrade pip
+    python3.7 -m pip install setuptools
+
+    python3.7 -m pip install -r source/Mlos.Python/requirements.txt
+    ```

```

Comment:
Dunno if this should be addressed here or as a separate PR but #31, #45, and #43 all mention additional improvements in Python dependency setup.

Reply:
Discussed with @amueller earlier.  I think we're going to leave it here for now and fix it up afterwards.

For now I left a section for Conda install instructions as well.